src/core/console.ts

```diff
@@ -1,24 +1,40 @@
-import filterConsole from 'filter-console'
+const consoleFilters = [
+  /^The above error occurred in the <.*?> component:/, // error boundary output
+  /^Error: Uncaught .+/ // jsdom output
+]
 
 function suppressErrorOutput() {
-  if (process.env.RHTL_DISABLE_ERROR_FILTERING) {
-    return () => {}
-  }
+  const originalError = console.error
 
-  return filterConsole(
-    [
-      /^The above error occurred in the <TestComponent> component:/, // error boundary output
-      /^Error: Uncaught .+/ // jsdom output
-    ],
-    {
-      methods: ['error']
+  const error = (...args: Parameters<typeof originalError>) => {
+    const message = typeof args[0] === 'string' ? args[0] : null
+    if (!message || !consoleFilters.some((filter) => filter.test(message))) {
```

Comment:
Just to call it out, `filter-console` would format the string before testing it against the patter (that's why it had a dependency on `util`), but I found this wasn't necessary for the cases we are trying to filter.  Not doing this may be an issue if how the calls from upstream are made ever change or if we introduce any new cases in the future (although I don't think that is likely).